### PR TITLE
Add support for Duration type

### PR DESCRIPTION
**Pull request by martin-sucha**

### CODE REVIEW

**Comment by martin-sucha** [comment]
related to https://github.com/scylladb/scylla-go-driver/issues/158

File: frame/vint_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
+package frame
+
+import (
+	"math"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestDecodeVint(t *testing.T) {
```
Comment (Kulezi) [inline frame/vint_test.go:10]: Tests in this file could check behavior for all 9 vint lengths

File: frame/vint_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		},
+		{
+			name:  "256000",
+			data:  []byte{0xc7, 0xd0, 0x00},
+			value: 256000,
+		},
+		{
+			name:  "123456789",
+			data:  []byte{0xee, 0xb7, 0x9a, 0x29},
+			value: -123456789,
```
Comment (Kulezi) [inline frame/vint_test.go:107]: is it positive or negative?

File: frame/cqlvalue_test.go [hunk 3]
```diff
@@ -909,3 +909,119 @@ func TestCqlValueStringMap(t *testing.T) {
...
+				Type: &Option{
+					ID: DurationID,
+				},
+				Value: Bytes{0x02, 0x04, 0x06},
+			},
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (martin-sucha) [reply frame/cqlvalue_test.go]: Yeah, I've generated it using IDE (GoLand). Thanks for noticing! Fixed.

File: frame/cqlvalue_test.go [hunk 4]
```diff
@@ -909,3 +909,119 @@ func TestCqlValueStringMap(t *testing.T) {
...
+			t.Parallel()
+			res := CqlFromDuration(tc.value)
+			if diff := cmp.Diff(res, tc.content); diff != "" {
+				t.Fatalf(diff)
+			}
+		})
+	}
+}
+
+func TestCqlValue_AsDuration(t *testing.T) { // nolint:dupl // Tests are different.
```
[the remark this answers is not in the document]
Comment (martin-sucha) [reply frame/cqlvalue_test.go]: I've generated the method using IDE and it put the underscore there automatically. Renamed.

File: frame/vint_test.go (hunk 2 above)
Comment (martin-sucha) [reply frame/vint_test.go]: Negative. Fixed the test name. Thanks!

File: frame/vint_test.go (hunk 1 above)
Comment (martin-sucha) [reply frame/vint_test.go:10]: Added.

File: frame/cqlvalue.go [hunk 5]
```diff
@@ -265,6 +265,36 @@ func (c CqlValue) AsStringMap() (map[string]string, error) {
...
+	}
+	raw = raw[n:]
+	nanoseconds, _, err := decodeVInt(raw)
+	return Duration{
+		Months:      int32(months),
+		Days:        int32(days),
+		Nanoseconds: nanoseconds,
+	}, err
+}
+
```
[the remark this answers is not in the document]
Comment (martin-sucha) [reply frame/cqlvalue.go]: Added a check here as well.

File: frame/cqlvalue.go [hunk 6]
```diff
@@ -265,6 +265,36 @@ func (c CqlValue) AsStringMap() (map[string]string, error) {
...
+		return Duration{}, fmt.Errorf("days out of range")
+	}
+	raw = raw[n:]
+	nanoseconds, _, err := decodeVInt(raw)
+	return Duration{
+		Months:      int32(months),
+		Days:        int32(days),
+		Nanoseconds: nanoseconds,
+	}, err
+}
```
[the remark this answers is not in the document]
Comment (Kulezi) [reply frame/cqlvalue.go]: Thanks for noticing, will fix that in the mentioned functions

**Comment by Kulezi** [comment]
LGTM, please address linter comments and this can be merged.

**Comment by martin-sucha** [comment]
Fixed.

**Pull request merged**